File: Java Script - Introduction to Programming/8.1. Exam Preparation - Part I/10. Rectangle with Stars.js
function rectangleWithStars(n) {

  //top
  console.log('%'.repeat(n * 2))

  //before middle
  for (let i = 0; i < n / 2 - 1; i++) {
    console.log('%' + ' '.repeat(n * 2 - 2) + '%')
  }

  //middle
  console.log('%' + ' '.repeat(n - 2) + '**' + ' '.repeat(n - 2) + '%')

  //after  middle
  for (let i = 0; i < n / 2 - 1; i++) {
    console.log('%' + ' '.repeat(n * 2 - 2) + '%')
  }

  //bottom
  console.log('%'.repeat(n * 2))
}

// rectangleWithStars(['2'])
// rectangleWithStars(['3'])

File: Pyrhon - Introduction to Programming/8.1. Exam Preparation - Part I/10. Rectangle with Stars.py
from math import ceil

n = int(input())
#top
print('%' * (n * 2))

#before middle
[print('%' + ' ' * (n * 2 - 2) + '%') for i in range(ceil(n / 2) - 1)]

#middle
print('%' + ' ' * (n - 2) + '**' + ' ' * (n - 2) + '%')

#after middle
[print('%' + ' ' * (n * 2 - 2) + '%') for i in range(ceil(n / 2) - 1)]

#bottom
print('%' * (n * 2))


'''
5
%%%%%%%%%%
%        %
%        %
%   **   %
%        %
%        %
%%%%%%%%%%
'''
